### PR TITLE
Restore integration tests

### PR DESCRIPTION
**Pull request by jaredhirsch**
Get our lone integration test working again on circle-CI.

TL;DR: mostly just needed to bump deps.

Some breaking changes occurred between Firefox and the Marionette python client. It looks like https://bugzil.la/1302707 (landed around the time #1437 was filed) removed `marionette.timeouts`, which broke the self-test that runs inside marionette 3.1.0. I was seeing this error locally today, not sure if others did or not. Upgrading marionette to 3.2.0 (le sigh, a minor version number bump for a breaking API change...) and updating firefox-puppeteer to latest seems to have fixed everything.

Also tweaked the integration test to stop looking for the now-defunct `/experiments` redirect, and cleaned up the docs and .gitignore a bit. :beers:

### CODE REVIEW

**Comment by dannycoates** [comment]
screen shot 2016-11-02 at 8 50 55 pm

👍 

squash maybe?

File: bin/circleci/run-integration-tests.sh [hunk 1]
```diff
@@ -12,7 +12,7 @@ done
 python integration/runtests.py \
     --binary=$HOME/firefox/firefox-bin \
     --verbose \
-    integration
+    integration/test_installation.py
```
Comment (lmorchard) [inline bin/circleci/run-integration-tests.sh:15]: If we add more test cases, do we have to list them all here?

Comment (jaredhirsch) [reply bin/circleci/run-integration-tests.sh:15]: I'm hoping to figure that out while I'm writing the second integration test :-)

Comment (jaredhirsch) [reply bin/circleci/run-integration-tests.sh:15]: Or rather, I'm hoping to figure out a working glob pattern while writing the second test

**Comment by jaredhirsch** [comment]
@dannycoates cool, thanks for looking. squashed

**Review by lmorchard — APPROVED** [review]
LGTM, will merge after this test run finishes
